### PR TITLE
Use code points instead of bytes for STRSUB/STRLEN

### LINKED ISSUE #297: Incorrect STRLEN
STRLEN() returns the wrong value for "NIDORAN♂".  It appears to return 10 instead of 8.  Maybe it is based on the byte length rather than the char length?

### LINKED ISSUE #348: STRSUB produces invalid UTF-8 character errors
lines such as `db STRSUB("ポ", 1, 1)` produce the error `invalid UTF-8 character` likely due to strsub only getting a single byte from the multibyte character.

### PR DESCRIPTION
**Pull request by dbrotz**
I've changed STRSUB/STRLEN to treat strings as being encoded in UTF-8 and to count in terms of code points rather than bytes, as we discussed in issue #348.

Fixes #297 and fixes #348.

### CODE REVIEW

File: test/asm/strsub.asm [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	xstrsub "ABC", 2, 1
+	xstrsub "ABC", 3, 1
+	xstrsub "ABC", 1, 2
+	xstrsub "ABC", 2, 2
+	xstrsub "ABC", 2, 32
+	xstrsub "ABC", 2, 300
+	xstrsub "ABC", 0, 300
+	xstrsub "ABC", 4, 0
+	xstrsub "ABC", 4, 1
+	xstrsub "カタカナ", 1, 2
```
Comment (AntonioND) [inline test/asm/strsub.asm]: Nice test string. xD

**Comment by AntonioND** [comment]
This looks good to me, but I'll leave it here for a bit for others to see before merging it.

**Comment by dbrotz** [comment]
Thanks.

**Comment by NieDzejkob** [comment]
The tests are failing on my machine:
```
Note: checking out 'd00ec024a25be214377f0b1449e6676053bf392b'.
~/.../test/asm$ ./test.sh
--- strlen.out  2019-07-03 15:38:14.016987681 +0200
+++ /tmp/tmp.R9AHl8Lwfj 2019-07-03 15:40:45.077308085 +0200
@@ -1,2 +1,2 @@
 $3
-$4
+$C
--- strsub.out  2019-07-03 15:38:14.016987681 +0200
+++ /tmp/tmp.R9AHl8Lwfj 2019-07-03 15:40:45.080641541 +0200
@@ -1,19 +1,7 @@
-warning: strsub.asm(13) -> xstrsub(1):
-    STRSUB: Length too big: 32
 warning: strsub.asm(14) -> xstrsub(1):
     STRSUB: Length too big: 300
 warning: strsub.asm(15) -> xstrsub(1):
-    STRSUB: Position starts at 1
-warning: strsub.asm(15) -> xstrsub(1):
     STRSUB: Length too big: 300
-warning: strsub.asm(16) -> xstrsub(1):
-    STRSUB: Position 4 is past the end of the string
-warning: strsub.asm(17) -> xstrsub(1):
-    STRSUB: Position 4 is past the end of the string
-warning: strsub.asm(17) -> xstrsub(1):
-    STRSUB: Length too big: 1
-warning: strsub.asm(20) -> xstrsub(1):
-    STRSUB: Length too big: 10
 A
 B
 C
@@ -21,11 +9,11 @@
 BC
 BC
 BC
-ABC


-カタ
-カナ
-カナ
+
+
+
+タカナ
 g
-g̈
+g
```

For the record, `strsub.asm`, as checked out by git for me, has this SHA-1:
`594ee5428dc697adc583a996232bda000ebc4c1b  strsub.asm`

**Pull request merged**